packages/SwingSet/misc-tools/replay-transcript.js

```diff
@@ -151,27 +248,48 @@ async function replay(transcriptFile) {
     }
     lineNumber += 1;
     const data = JSON.parse(line);
-    if (!manager) {
+    if (data.type === 'heap-snapshot-load') {
```

Comment:
Do we have anything that creates a `heap-snapshot-load` transcript entry?

For most purposes, I think we'd be replaying from the most recent snapshot (so we might `save` snapshots during the course of the replay, but we'd never `load` more than one). Our extract-from-slogfile tool might grow to let the operator select which span of the vat they're exercising, and if the slogfile only goes back to the most recent kernel restart, then that span might start from a snapshot instead of from the initial vat state). The extract-from-kerneldb tool could would know the most recent snapshot and transcript since then, but it would not know about `save`s (but that'd be fine, since by definition no saves happen during the transcript-suffix: if one did, the starting point would be moved to exclude it).

Reply:
Not yet, but I wanted to expand the slog extraction tool to support concatenated or multiple slog files, or in general paging out and reloading a vat. Currently it chokes on any "replay" which come with snapshot load (for lack of syscall results in the slog), but we could paper over these if we've already seen the same deliveries.